### PR TITLE
Nix + macOS

### PR DESCRIPTION
**Pull request by zimbatm**
A few changes to allow Nix usage on macOS.

The last remaining issue is that snack doesn't know how to link properly on macOS /cc @nmattia

### CODE REVIEW

**Comment by nmattia** [comment]
@zimbatm can you clarify what you mean by "snack doesn't know how to link" ? Snack's CI passes on OSX: https://travis-ci.org/nmattia/snack/jobs/387982194#L1152, which does simple linking of the test executables.

File: nix/build-support/reflex/default.nix [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{ buildGoPackage, fetch }:
```
Comment (nmattia) [inline nix/build-support/reflex/default.nix]: let's keep the convention of `nix/<project>/[overlay.nix|src.json]`, unless there's a compelling reason not to do so.

Comment (zimbatm) [reply nix/build-support/reflex/default.nix]: like this? 97d71ef37b2c4d15e59b1559926eb25d46ea5588

**Comment by zimbatm** [comment]
Come to think of it, I'm not sure if nix is running in sandboxed mode on Travis

**Comment by zimbatm** [comment]
in the end there was only one commit to keep. macOS still doesn't work entirely on Travis CI

**Pull request merged**